### PR TITLE
New vertical updraft velocity method

### PR DESCRIPTION
**Pull request by jkreschke**
Implemented implicit method for clubb mf upw equation. Added logical option to use implicit method, with default value false

### CODE REVIEW

File: src/physics/cam/clubb_mf.F90 [hunk 1]
```diff
@@ -628,17 +632,23 @@ subroutine integrate_mf( nz,
              end if
            end if
 
-           ! get wn^2
-           wp = wb*ent(k+1,i)*eturb
-           if (wp==0._r8) then
-             wn2 = upw(k,i)**2._r8+2._r8*wa*B*dzt(k+1)
+           if (do_implicit) then
+             wp = clubb_mf_alphturb*wb*ent(k+1,i)*sqrt(tke(k+1))*dzt(k+1)
```
Comment (adamrher) [inline src/physics/cam/clubb_mf.F90]: @jkreschke I think the TKE belongs at the midpoint levels for the implicit method. Only the w needs to be taken at k+1 on the RHS, I think. If you agree can you change to 0.5_r8*(tke(k+1)+tke(k))?

File: src/physics/cam/clubb_mf.F90 [hunk 2]
```diff
@@ -628,17 +632,23 @@ subroutine integrate_mf( nz,
              end if
            end if
 
-           ! get wn^2
-           wp = wb*ent(k+1,i)*eturb
-           if (wp==0._r8) then
-             wn2 = upw(k,i)**2._r8+2._r8*wa*B*dzt(k+1)
+           if (do_implicit) then
+             wp = clubb_mf_alphturb*wb*ent(k+1,i)*sqrt(tke(k+1))*dzt(k+1)
+             wn = (-wp + sqrt(wp**2._r8 + (1._r8 + 2._r8*wb*ent(k+1,i)*dzt(k+1))*(upw(k,i)**2._r8 + 2._r8*wa*B*dzt(k+1))) )/(1._r8 + 2._r8*wb*ent(k+1,i)*dzt(k+1))
```
Comment (adamrher) [inline src/physics/cam/clubb_mf.F90]: can you split this into two lines using the & sign please?

**Comment by jkreschke** [comment]
Done. The implicit method now uses TKE at the midpoint.

**Review by mchinita — APPROVED** [review]
I don't see any issues. Thanks!

**Review by adamrher — APPROVED** [review]
Looks good. Thanks Jake!

**Pull request merged**